server/server.go

```diff
@@ -126,7 +126,7 @@ func NewServer(conf *config.Config, logger log.Logger) (*Server, error) {
 
 	var proxyVerifier *auth.MultiTenantVerifier
 	if conf.Proxy.Auth.Enabled() {
-		verifierConf, err := conf.Proxy.Auth.Load()
+		verifierConf, err := conf.Proxy.Auth.Load(s.rebalanceCtx)
```

Comment:
I'm not sure `rebalanceCtx` is the right chose here? It will be closed before all servers are closed

It doesn't really matter if the background goroutine keeps running until the process exits? So can just use a background context to keep simple

Reply:
See what you think of this: https://github.com/andydunstall/piko/pull/267/commits/9cf4a5e62714b77d40e229f010991b59b343b541
Potentially an overkill, but maybe good to have anyway?